lms/djangoapps/grades/management/commands/reset_grades.py

```diff
@@ -0,0 +1,151 @@
+
+from lms.djangoapps.grades.models import PersistentSubsectionGrade, PersistentCourseGrade
+
+
+log = logging.getLogger(__name__)
+
+
+DATE_FORMAT = "%Y-%m-%d %H:%M"
+
+
+class Command(BaseCommand):
+    """
+    Reset persistent grades for learners.
+    """
+    help = dedent(__doc__).strip()
+
+    def add_arguments(self, parser):
+        """
+        Add arguments to the command parser.
+        """
+        parser.add_argument(
+            '--dry_run',
+            action='store_true',
+            default=False,
+            dest='dry_run',
+            help="Output what we're going to do, but don't actually do it. To actually delete, use --delete instead."
+        )
+        parser.add_argument(
+            '--delete',
+            action='store_true',
+            default=False,
+            dest='delete',
+            help="Actually perform the deletions of the course. For a Dry Run, use --dry_run instead."
+        )
+        parser.add_argument(
+            '--courses',
+            dest='courses',
+            nargs='+',
+            help='Reset persistent grades for the list of courses provided.',
+        )
+        parser.add_argument(
+            '--all_courses',
+            action='store_true',
+            dest='all_courses',
+            default=False,
+            help='Reset persistent grades for all courses.',
+        )
+        parser.add_argument(
+            '--modified_start',
+            dest='modified_start',
+            help='Starting range for modified date (inclusive): e.g. "2016-08-23 16:43"',
+        )
+        parser.add_argument(
+            '--modified_end',
+            dest='modified_end',
+            help='Ending range for modified date (inclusive): e.g. "2016-12-23 16:43"',
+        )
+
+    def handle(self, *args, **options):
+        course_keys = None
+        modified_start = None
+        modified_end = None
+
+        run_mode = self._get_mutually_exclusive_option(options, 'delete', 'dry_run')
+        courses_mode = self._get_mutually_exclusive_option(options, 'courses', 'all_courses')
+
+        if options.get('modified_start'):
+            modified_start = datetime.strptime(options['modified_start'], DATE_FORMAT)
+
+        if options.get('modified_end'):
+            if not modified_start:
+                raise CommandError('Optional value for modified_end provided without a value for modified_start.')
+            modified_end = datetime.strptime(options['modified_end'], DATE_FORMAT)
+
+        if courses_mode == 'courses':
+            try:
+                course_keys = [CourseKey.from_string(course_key_string) for course_key_string in options['courses']]
+            except InvalidKeyError as error:
+                raise CommandError('Invalid key specified: {}'.format(error.message))
+
+        log.info("reset_grade: Started in %s mode!", run_mode)
+
+        operation = self._query_grades if options.get('dry_run') else self._delete_grades
```

Comment:
Nit: use the already-parsed `run_mode` variable here in your if clause, instead of going all the way back to `options.get('dry_run')`